### PR TITLE
Issue #86: Auto-capitalize off problems

### PR DESCRIPTION
**Pull request by michalhasko**

### CODE REVIEW

File: app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt [hunk 1]
```diff
@@ -241,13 +243,17 @@ fun getImeActionCode(ime: IMEService): Int {
 /**
  * Returns the correct keyboard mode
  */
-fun getKeyboardMode(ime: IMEService): KeyboardMode {
+fun getKeyboardMode(ime: IMEService): (AppSettings?) -> KeyboardMode {
```
Comment (michalhasko) [inline app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt:246]: I don't know how to work properly with `@Composable`, so I couldn't extract the app settings in `IMEService` to pass `autoCapitalize` it here. So I made it to return a lambda.

File: app/src/main/java/com/dessalines/thumbkey/ComposeKeyboardView.kt [hunk 2]
```diff
@@ -15,7 +15,7 @@ import com.dessalines.thumbkey.utils.KeyboardMode
 class ComposeKeyboardView(
     context: Context,
     private val liveSettings: LiveData<AppSettings>,
-    private val startMode: KeyboardMode
+    private val startMode: (AppSettings) -> KeyboardMode
```
Comment (dessalines) [inline app/src/main/java/com/dessalines/thumbkey/ComposeKeyboardView.kt:18]: Uhhh what's going on here? This problem doesn't need a lambda solution.

File: app/src/main/java/com/dessalines/thumbkey/ComposeKeyboardView.kt [hunk 3]
```diff
@@ -27,7 +27,7 @@ class ComposeKeyboardView(
         ThumbkeyTheme(
             settings = settings
         ) {
-            KeyboardScreen(settings, startMode)
+            KeyboardScreen(settings, startMode(settings as AppSettings))
```
Comment (dessalines) [inline app/src/main/java/com/dessalines/thumbkey/ComposeKeyboardView.kt:30]: Why are you changing the type of settings?

File: app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt (hunk 1 above)
Comment (dessalines) [reply app/src/main/java/com/dessalines/thumbkey/utils/Utils.kt:246]: This  is a function. You pass in parameters by using function parameters. So provide it with either the settings, or settings?.autocapitalize.

**Comment by michalhasko** [comment]
I'm clearly out of my depth, closing the PR.

**Pull request closed without merging**